aea/protocols/dialogue/base.py

```diff
@@ -1071,6 +1071,13 @@ def __init__(self, dialogues: "Dialogues") -> None:
         # used for both storing complete and incomplete dialogue labels
         self._terminal_state_dialogues_labels: Set[DialogueLabel] = set()
 
+    def cleanup(self) -> None:
```

Comment:
I just noticed that BaseDialoguesStorage has a dependency on Dialogues which means there is circularity. Can we break this? Quick check seems to suggest we don't need self._dialogues here (apart from a single flag, which can be provided as an argument to BaseDialoguesStorage instead).

Reply:
`PersistDialoguesStorage` (derived from `BasicDialoguesStorage`) uses `self._dialogues`  in several places, but I'll try to break the dependency.

Reply:
I think it would be better to try this on a separate PR and roll back the changes here!